bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/AccessibleToolBar.java

```diff
@@ -35,11 +34,14 @@
  */
 class AccessibleToolBar extends Composite {
 
-	private List<ToolBar> toolBars = new ArrayList<>();
+	private final List<AccessibleToolItem> accessibleToolItems = new ArrayList<>();
+
+	private final GridLayout layout;
 
 	public AccessibleToolBar(Composite parent) {
 		super(parent, SWT.NONE);
-		GridLayoutFactory.fillDefaults().numColumns(0).spacing(0, 0).margins(0, 0).applyTo(this);
+		this.layout = GridLayoutFactory.fillDefaults().numColumns(0).spacing(0, 0).margins(0, 0).create();
```

Comment:
Storing the layout improves static correctness, as you do not need to call `getLayout()` and optimistically cast it to `GridLayout` anymore but have the guarantee that it has the correct type. In case someone changes the code, it will fail statically (instead of having a `ClassCastException` at runtime).

Reply:
Thank you for the insight, I will keep this in mind in the future :)